java/integTest/src/test/java/glide/PubSubTests.java

```diff
@@ -890,4 +879,293 @@ public void transaction_with_all_types_of_PubsubMessages(
                                 Pair.of(1, exactMessage), Pair.of(1, patternMessage), Pair.of(1, shardedMessage));
         verifyReceivedPubsubMessages(expected, listener, useCallback);
     }
+
+    @SneakyThrows
+    @ParameterizedTest(name = "standalone = {0}")
+    @ValueSource(booleans = {true, false})
+    @Disabled(
+            "No way of currently testing this, see https://github.com/aws/glide-for-redis/issues/1649")
+    public void pubsub_exact_max_size_message(boolean standalone) {
+        final GlideString channel = gs(UUID.randomUUID().toString());
+        final GlideString message = gs("1".repeat(1 << 25)); // 33MB
+        final GlideString message2 = gs("2".repeat(1 << 25)); // 3MB
+
+        Map<? extends ChannelMode, Set<GlideString>> subscriptions =
+                standalone
+                        ? Map.of(PubSubChannelMode.EXACT, Set.of(channel))
+                        : Map.of(PubSubClusterChannelMode.EXACT, Set.of(channel));
+        var listener = createClientWithSubscriptions(standalone, subscriptions);
+        var sender = createClientWithSubscriptions(standalone, subscriptions);
+
+        try {
+            assertEquals(OK, sender.publish(message, channel).get());
+            assertEquals(OK, sender.publish(message2, channel).get());
+
+            // Allow the message to propagate.
+            Thread.sleep(MESSAGE_DELIVERY_DELAY);
+
+            PubSubMessage asyncMessage = listener.getPubSubMessage().get();
+            PubSubMessage syncMessage = listener.tryGetPubSubMessage();
+            assertEquals(message, asyncMessage.getMessage());
+            assertEquals(channel, asyncMessage.getChannel());
+            assertTrue(asyncMessage.getPattern().isEmpty());
+
+            assertEquals(message2, syncMessage.getMessage());
+            assertEquals(channel, syncMessage.getChannel());
+            assertTrue(syncMessage.getPattern().isEmpty());
+
+            // Assert there are no more messages to read.
+            assertThrows(
+                    TimeoutException.class,
+                    () -> {
+                        listener.getPubSubMessage().get(3, TimeUnit.SECONDS);
+                    });
+            assertNull(listener.tryGetPubSubMessage());
+        } finally {
+            if (!standalone) {
```

Comment:
There is `cleanup` method which does that after every test. Please remove in all tests.